Update node.js to 8.10.0, the latest LTS version (BL-5777)

StephenMcConnel opened the pull request:
This merge requires the affected TeamCity builds to switch to using node 8.10.0 before it will build on them.  The packaging builds on Jenkins will build okay since the code checked in here changes that behavior for those builds.

The new package-lock.json may or may not need to be checked in.  The discussion about it was not unanimous by any means.

Do not merge this if you aren't will to modify the TeamCity builds.  Of course, if you don't modify those builds first, how will you know it works?  We have a chicken and egg problem in one sense...

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bloombooks/bloomdesktop/2313)
<!-- Reviewable:end -->

StephenMcConnel commented:
I did try this on both Linux and Windows, and everything seemed to work okay, including "npm run test".

JohnThomson commented:
Steve, were you able to determine how upgrading TC to Node 8 will affect our ability to build new versions of Bloom 4.1 (and, of course, any other things that are built on TC and nuse Node :-)?

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/bloombooks/bloomdesktop/2313#-:-L8gk_LE5616ql-tbPql:b4cglh1)*
<!-- Sent from Reviewable.io -->

StephenMcConnel commented:
The TC builds individually use nvm to set the node version.  So I don't expect builds of 4.1 to be affected at all if we change the builds of 4.2 and master to node 8.
Similarly, the Jenkins package builds of Bloom each explicitly download the version of node they want, so older builds would not be affected by newer versions changing to node 8.

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/bloombooks/bloomdesktop/2313#-:-L8gsr7v1w_pVBE2Z6c_:bx8k2es)*
<!-- Sent from Reviewable.io -->

andrew-polk commented:
> The new package-lock.json may or may not need to be checked in. The discussion about it was not unanimous by any means.

What discussion? How do we make a decision?

andrew-polk commented:
Per discussion, without a compelling reason to change our process, let's do what we have been doing: use npm-shrinkwrap before deploying a beta version, and do not use package-lock.

Also, let's see how easy we can make it on devs to switch branches when a different version of npm is used (at least update documentation).

andrew-polk commented:
Is it worth adding a note in the readme about npm-build.sh?
Otherwise, <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/bloombooks/bloomdesktop/2313#-:-L8i2T5H24FtH2mjF8Rj:b-ygwb81)*
<!-- Sent from Reviewable.io -->

StephenMcConnel commented:
I added a note in the readme, and also moved and enhanced the script.  I decided to move it when I saw that it was being copied to output/browser.

---

Review status: 3 of 5 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/bloombooks/bloomdesktop/2313#-:-L8iIYLyDF-FHwxcarMg:bw07rwc)*
<!-- Sent from Reviewable.io -->

andrew-polk commented:
Reviewed 2 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/bloombooks/bloomdesktop/2313)*
<!-- Sent from Reviewable.io -->
